tests/ttnn/unit_tests/gtests/tensor/test_vector_conversion.cpp

```diff
@@ -0,0 +1,119 @@
+// SPDX-FileCopyrightText: (c) 2024 Tenstorrent AI ULC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include <gtest/gtest.h>
+#include <gmock/gmock.h>
+#include <algorithm>
+#include <cstdint>
+
+#include "ttnn/tensor/tensor.hpp"
+#include "ttnn/tensor/xtensor/conversion_utils.hpp"
+#include "ttnn/tensor/xtensor/xtensor_all_includes.hpp"
+
+namespace ttnn {
+namespace {
+
+using ::testing::Eq;
+using ::testing::Pointwise;
+
+const std::vector<ttnn::SimpleShape>& get_shapes_for_test() {
+    static auto* shapes = new std::vector<ttnn::SimpleShape>{
+        ttnn::SimpleShape{1},
+        ttnn::SimpleShape{1, 1, 1, 1},
+        ttnn::SimpleShape{1, 1, 1, 10},
+        ttnn::SimpleShape{1, 32, 32, 16},
+        ttnn::SimpleShape{1, 40, 3, 128},
+        ttnn::SimpleShape{2, 2},
+        ttnn::SimpleShape{1, 1, 1, 1, 10},
+    };
+    return *shapes;
+}
+
+TensorSpec get_tensor_spec(const ttnn::SimpleShape& shape, DataType dtype, Layout layout = Layout::ROW_MAJOR) {
+    return TensorSpec(shape, TensorLayout(dtype, layout, MemoryConfig{}));
+}
+
+template <typename T>
+std::vector<T> arange(int64_t start, int64_t end, int64_t step) {
+    std::vector<T> result;
+    for (int el : xt::arange<int64_t>(start, end, step)) {
```

Comment:
I wonder, does xtensor allow for a parallel arange?

Reply:
I don't think so...